### DISCUSSION TITLE
Link prediction on a new graph

### DISCUSSION THREAD

**Discussion by maurovit** [stellargraph/stellargraph#1841] [Q&A]
I trained a model for link classification using GraphSAGE and then I saved it. My goal is to load the model and use it to make predictions about links in a new graph that I did not use during training. In particular, I have only the nodes of this new graph and I have to find out the edges between these nodes based on the previous training. How can I do this?

**Comment by CalmLogarithm** [comment]
Have you read the [StellarGraph demo on GraphSAGE link prediction](https://stellargraph.readthedocs.io/en/stable/demos/calibration/calibration-link-prediction.html)? It both learns node embeddings and link prediction in parallel.

Once you have that model trained you can apply it to a graph. If you are adding the n+1 node to a graph, you would run the link prediction model against the n other nodes already in the graph.

However, there is a more fundamental question. You train GraphSAGE on one graph and then apply it to another graph. Is the first graph representative of the data you will see in the second graph? Are you training it on something that is relevant to the problem?
